rfcs/InputUnion.md

```diff
@@ -149,6 +149,13 @@ There have been a variety of use cases described by users asking for an abstract
 * [Observability Cloud Integrations](https://gist.github.com/binaryseed/f2dd63d1a1406124be70c17e2e796891#cloud-integrations)
 * [Observability Dashboards](https://gist.github.com/binaryseed/f2dd63d1a1406124be70c17e2e796891#dashboards)
 
+## Prior art
```

Comment:
It's more background information than prior art but it's probably worth linking to some general information on [structural typing vs. nominal typing](http://wiki.c2.com/?NominativeAndStructuralTyping) since that's pretty core to the input union problem.